Iceberg Plugin.

Parth-Brahmbhatt opened the pull request:
Initial patch with Iceberg plugin (https://github.com/Netflix/iceberg). The current implementation allows create, CTAS, drop, Select with predicate pushdown and some schema manipulation like add/remove columns. There are still more operations that needs to be supported and some of them require extending the parser but I already have a pretty huge PR. 

Before anyone starts reviewing I would like to point out that one big flaw in the current implementation is that I had to copy the parquet package from presto-hive to presto-iceberg. Iceberg requires a newer version of parquet which has resulted in renaming of all parquet class as parquet was moved to apache. parquet code was recently moved to its own module in presto-parquet which is a step in the right direction and I can simplify this PR and remove all duplication if we can bump the parquet version.

dain commented:
I'm a bit confused about what Iceberg is.  The Iceberg documentation refers to this as a Hive table format, but this is a completely new connector which would imply that it is a replacement for Hive?  Or put another way, why isn't this just part of the Hive plugin?

dain commented:
@zhenxiao and @nezihyigitbasi, do you see any issues with upgrading the Parquet dep?

nezihyigitbasi commented:
I remember trying to update Parquet quite some time ago and had plenty of issues while doing that. I couldn't even come to the point of hitting runtime issues at that time. Having said that we are at 1.6 and moving to 1.10 will bring lots of changes (see [this](https://github.com/apache/parquet-mr/blob/master/CHANGES.md) doc). It has been a while since we haven't updated Parquet, and now since there is this iceberg connector I think it makes sense to update & verify it. Since Netflix is a big Parquet user I hope @Parth-Brahmbhatt can help us with that :)

zhenxiao commented:
from 1.6 to 1.10 is a big change, better we separate parquet upgrade to another commit. try upgrade Parquet in Presto shaded jar?

Parth-Brahmbhatt commented:
@dain Iceberg is a table format and does not support tracking of tables in schema and it relies on some subsystem that supports atomic swap of table metadata location. The easiest approach was to use an existing metastore so I went with that and added a HiveMetastore based implementation in iceberg. My first intuition was also to try and just add Iceberg implementation in hive plugin but the code basically becomes a huge mess because a lot of constructs don't have counterparts in the hive and vice versa and it just felt like I was trying to force fit them.

@nezihyigitbasi I will try the version upgrade myself and see if I can address the issues.

@zhenxiao I am relying on iceberg-presto-runtime shaded jar to pull in the parquet classes it requires right now. It did not require additional shading because from 1.6 to 1.10 all package names changed in parquet from `parquet.*` to `org.apache.parquet.*`.  Are you suggesting something else? 

@ilfrin Sure I can break it down. I prefer to upgrade parquet version first and then just post the module which should look a lot smaller. If not we can divide it in smaller patches. I have already rebased the changes with master.

ilfrin commented:
Great, thank you @Parth-Brahmbhatt :)
